substrate-node/pallets/pallet-tfgrid/src/tests.rs

```diff
@@ -705,6 +705,93 @@ fn update_node_moved_from_farm_list_works() {
     });
 }
 
+#[test]
+fn update_certified_node_resources_loses_certification_works() {
+    ExternalityBuilder::build().execute_with(|| {
+        create_entity();
+        create_twin();
+        create_farm();
+        create_node();
+
+        assert_ok!(TfgridModule::add_node_certifier(
+            RawOrigin::Root.into(),
+            alice()
+        ));
+
+        assert_ok!(TfgridModule::set_node_certification(
+            Origin::signed(alice()),
+            1,
+            NodeCertification::Certified
+        ));
+
+        let node = TfgridModule::nodes(1).unwrap();
+        assert_eq!(node.certification, NodeCertification::Certified);
+
+        // Change cores to 2
+        let mut node_resources = node.resources;
+        node_resources.cru = 2;
+
+        assert_ok!(TfgridModule::update_node(
+            Origin::signed(alice()),
+            1,
+            1,
+            node_resources,
+            node.location,
+            node.country,
+            node.city,
+            Vec::new().try_into().unwrap(),
+            true,
+            true,
+            "some_serial".as_bytes().to_vec(),
+        ));
+
+        let node = TfgridModule::nodes(1).unwrap();
+        assert_eq!(node.certification, NodeCertification::Diy);
```

Comment:
Would be good to check the events too no?

Reply:
Could be done, but not really needed here since we know the actual field was set

Reply:
I agree, if not already checked somewhere else
And after quick check it is not

Reply:
Alright, let's add it then :)

Reply:
It doesn't hurt and we never know when we might start using the event in other places.